src/http/http2_types.h

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/http_header_map.h"
+#include "ccf/http_status.h"
+#include "ccf/rest_verb.h"
+#include "http_builder.h"
+
+#include <list>
+#include <memory>
+#include <nghttp2/nghttp2.h>
+#include <string>
+#include <vector>
+
+namespace http2
+{
+  using StreamId = int32_t;
+
+  constexpr static size_t max_data_read_size = 1 << 20;
```

Comment:
Just in case: if this is the maximum size of any request, we may want that to be a setting in the config, so that operators can adjust it if necessary.